src/main/java/graphql/annotations/processor/retrievers/GraphQLAdditionalTypesHandler.java

```diff
@@ -0,0 +1,116 @@
+public class GraphQLAdditionalTypesHandler {
+
+    private GraphQLInterfaceRetriever graphQLInterfaceRetriever;
+    private GraphQLObjectInfoRetriever graphQLObjectInfoRetriever;
+    private SearchAlgorithm methodSearchAlgorithm;
+    private SearchAlgorithm fieldSearchAlgorithm;
+    private GraphQLObjectInfoRetriever infoRetriever;
+    private ClassFinder classFinder;
+
+    public GraphQLAdditionalTypesHandler() {
+        classFinder = new ClassFinder();
+        classFinder.addClassPath();
+        infoRetriever = new GraphQLObjectInfoRetriever();
+    }
+
+    public Set<GraphQLType> getAdditionalInterfacesImplementations(Class<?> root, ProcessingElementsContainer container) {
+        Set<GraphQLType> additionalFields = new HashSet<>();
+
+        additionalFields.addAll(getNewImplementations(container, classFinder, root));
+        for (Method method : graphQLObjectInfoRetriever.getOrderedMethods(root)) {
+            if (method.isBridge() || method.isSynthetic()) {
+                continue;
+            }
+            if (methodSearchAlgorithm.isFound(method)) {
+                additionalFields.addAll(getAdditionalInterfacesImplementations(method.getReturnType(), container));
+            }
+        }
+
+        for (Field field : getAllFields(root).values()) {
+            if (Modifier.isStatic(field.getModifiers())) {
+                continue;
+            }
+            if (fieldSearchAlgorithm.isFound(field)) {
+                additionalFields.addAll(getAdditionalInterfacesImplementations(field.getType(), container));
+            }
+        }
+
+        return additionalFields;
+    }
+
+    private Set<GraphQLType> getNewImplementations(ProcessingElementsContainer container, ClassFinder classFinder, Class<?> aClass) {
+        Set<GraphQLType> additionalFields = new HashSet<>();
+        if (container.getTypeRegistry().containsKey(infoRetriever.getTypeName(aClass)) && aClass.isInterface() && aClass.isAnnotationPresent(GraphQLTypeResolver.class)) {
+            ClassFilter classFilter = new SubclassClassFilter(aClass);
+            List<ClassInfo> foundClasses = new ArrayList<>();
+            classFinder.findClasses(foundClasses, classFilter);
+            for (ClassInfo classInfo : foundClasses) {
+                try {
+                    if (!(container.getTypeRegistry().containsKey(infoRetriever.getTypeName(Class.forName(classInfo.getClassName()))))) {
+                        GraphQLOutputType additionalObject = graphQLInterfaceRetriever.getInterface(Class.forName(classInfo.getClassName()), container);
+                        additionalFields.add(additionalObject);
+                    }
+                } catch (ClassNotFoundException e) {
+                    throw new GraphQLException(e);
+                }
+            }
+        }
+        return additionalFields;
+    }
+
+    public void setGraphQLObjectInfoRetriever(GraphQLObjectInfoRetriever graphQLObjectInfoRetriever) {
```

Comment:
The @Reference annotations and the unsetter will be needed to correctly initialize the osgi component

Reply:
Can you tell me exactly what to do? I don't know anything in OSGI